lib/annotate/annotate_models.rb

```diff
@@ -249,7 +249,7 @@ def get_schema_info(klass, header, options = {})
       cols = cols.sort_by(&:name) if options[:sort]
       cols = classified_sort(cols) if options[:classified_sort]
       cols.each do |col|
-        col_type = (col.type || col.sql_type).to_s
+        col_type = ((col.type && col.type.type) || (col.sql_type && col.sql_type.to_s))
```

Comment:
how about
```ruby
col_type = (col.type || col.sql_type)
col_type = col_type.type if col_type.respond_to?(:type)
col_type= col_type.to_s
```

Because the one above doesn't look like it will continue working for older Rails versions